test/e2e/operator_test.go

```diff
@@ -3508,11 +3509,32 @@ func TestLoadBalancingAlgorithmUnsupportedConfigOverride(t *testing.T) {
 func TestUnsupportedConfigOverride(t *testing.T) {
 	t.Parallel()
 
+	var (
+		dcmDefaultValue         = ""
+		dcmUpdateValue          = "true"
+		dcmExpectedValue        = "true"
+		maxServersDefaultValue  = ""
+		maxServersUpdateValue   = "2"
+		maxServersExpectedValue = "" // max dynamic servers cannot be set if DCM is off
+	)
+
+	if dcmEnabled, err := isFeatureGateEnabled(features.FeatureGateIngressControllerDynamicConfigurationManager); err != nil {
+		t.Fatalf("failed to get dynamic config manager feature gate: %v", err)
+	} else if dcmEnabled {
+		dcmDefaultValue = "true"
```

Comment:
You may want to log something here to let us know this was run on a techpreview cluster.
```suggestion
                t.Logf("DynamicConfigurationManager feature gate is enabled for this test")
		dcmDefaultValue = "true"
```